AppCenter/AppCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -28,18 +31,22 @@ - (instancetype)initWithSchema:(MSDBSchema *)schema version:(NSUInteger)version
       }
       [MSDBStorage enableAutoVacuumInOpenedDatabase:db];
       [MSDBStorage setVersion:version inOpenedDatabase:db];
-      return SQLITE_OK;
-    }];
+      sqlite3_close(db);
+    };
   }
   return self;
 }
 
 - (int)executeQueryUsingBlock:(MSDBStorageQueryBlock)callback {
   int result;
-  sqlite3 *db = [self openDatabaseAtFileURL:self.dbFileURL withMaxPageCount:self.maxPageCount withResult:&result];
+  sqlite3 *db = [MSDBStorage openDatabaseAtFileURL:self.dbFileURL withResult:&result];
   if (!db) {
     return result;
   }
+
+  // The value is stored as part of the database connection and must be reset every time the database is opened.
+  long maxPageCount = self.maxSizeInBytes / self.pageSize;
```

Comment:
Do we really want to do the calculation every time we open the DB?

Reply:
At first I got page size every time and at that moment it made sense =) but then I started to cache the page size because it cannot be changed on existing database.

Reply:
Actually, one integer division operation doesn't make any impact on performance. So, it's not big deal.
But we are setting and operating with size in bytes, so storing `maxSizeInBytes` property is more useful than some **internal** value.

Reply:
I guess it's fine since we don't use the result anywhere else.